### DISCUSSION TITLE
Calculation of decay heat?

### DISCUSSION THREAD

**Discussion by aairfccha** [radioactivedecay/radioactivedecay#89] [Ideas]
I have recently discovered this neat package and currently try to learn to use it. 

It can be quite useful to know how much decay heat a given inventory generates, both at some point in time and over time. Could this feature be added to the package (or did I just miss it in the documentation)? Naively thought, with the activity already available as an output, this should be a simple multiplication with a constant factor (for each isotope involved) for the amount of heat generated per activity.

**Comment by alexmalins** [comment]
Hi @aairfccha 

> Naively thought, with the activity already available as an output, this should be a simple multiplication with a constant factor (for each isotope involved) for the amount of heat generated per activity.

Yes you're right, you just multiply the activity of each nuclide by the energy of the radiations produced from its decay to get the decay heat. 

> and over time.

To get the total (cumulative) heat over time, same as above but with the `cumulative_decays()` method which gives the total number of decays over a time period.


Currently this package does not contain data for the energy of the radiations produced by radioactive decay. You can find this data easily online e.g. [NuDat2](https://www.nndc.bnl.gov/nudat3/nudat2.jsp) or in the [ICRP-107 data files](https://journals.sagepub.com/doi/suppl/10.1177/ANIB_38_3). In future I hope to find time to include this data in this package, so you can make decay heat calculations and extract other radiation energy data easily in one place.

**Comment by aairfccha** [comment]
When calculating decay heat (semi-)manually, I ran into a bit of a GIGO issue. For example for Np-239, ICRP-107 gives a decay energy of 446.9 keV, the IAEA live chart of nuclides 398.7 keV (plus 278 keV as a neutrino which should be thermally irrelevant at normal scales) and ENDF/B.VIII 405.16 keV (plus 271.9 keV neutrino). When Np-239 is a major part of the decay power you want to calculate in the first place, 10% difference has a bit of an impact on the overall result. 

Unfortunately while the IAEA live chart has an API to access data directly, for this information each nuclide and each type of radiation apparently has to be queried separately. The ENDF-6 file format is difficult to decrypt for the most part and the human-readable section I got the data from is probably not trivial to parse automatically. ICRP-107 is the most accessible thanks to the big table of energies in the document, but with this sort of difference I don't really know how far to trust it.

**Reply by alexmalins** [reply]
ICRP-107 is based on ENDSF 2001 IIRC, so in this situation I'd use the newer data sources - either ENDF/B.VIII or the IAEA live charts, which is based on a more recent ENDSF, which itself is based on ENDF I think.

You could investigate the origin of the difference by looking at the radiations files in the ICRP-107 dataset, and comparing to ENDF 8 or [ENDSF](https://www.nndc.bnl.gov/nudat3/decaysearchdirect.jsp?nuc=239NP&unc=nds) numbers.

> 10% difference has a bit of an impact on the overall result.
> but with this sort of difference I don't really know how far to trust it.

For more of my thoughts on uncertainties check here: https://radioactivedecay.github.io/overview.html#limitations
